Client/Frontend/Browser/TabManager.swift

```diff
@@ -491,7 +491,27 @@ class TabManager: NSObject {
     if flushToDisk && !zombie && !isPrivate {
       saveTab(tab, saveOrder: true)
     }
-
+    
+    // When the state of the page changes, we debounce a call to save the screenshots and tab information
+    // This fixes pages that have dynamic URL via changing history
+    // as well as regular pages that load DOM normally.
+    var debounce_timer: Timer?
+    tab.onPageReadyStateChanged = { [weak self] state in
+      guard let self = self else { return }
+      
+      debounce_timer?.invalidate()
+      debounce_timer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) { _ in
+        debounce_timer?.invalidate()
+        
+        if state == .complete || state == .loaded || state == .pushstate || state == .popstate {
+          // Saving Tab Private Mode - not supported yet.
```

Comment:
just curious do we plan to support it? seems it'd need a privacy review if so

Reply:
In the future yes we might support it (or so I've heard), but it is a long way from now I think. I put the comment here because we don't currently support saving private tabs to disk. I think we will when we move to Chromium's Tabs implementation along with TabGroups and so on. That will require us to refactor anywhere with `Tab` (including the above). We will definitely open a security and privacy review if/when we do support it.

Reply:
sgtm